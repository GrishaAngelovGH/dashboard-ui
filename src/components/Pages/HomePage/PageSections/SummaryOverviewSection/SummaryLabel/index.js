import SummaryLabel from './SummaryLabel'

export default SummaryLabel