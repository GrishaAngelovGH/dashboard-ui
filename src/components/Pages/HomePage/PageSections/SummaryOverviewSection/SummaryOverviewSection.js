import { Component } from 'react'
import PropTypes from 'prop-types'

import LinkButton from 'components/LinkButton'
import PageSection from 'components/Pages/PageSection'
import SummaryLabel from './SummaryLabel'

import { WOW } from 'wowjs'

class SummaryOverviewSection extends Component {
    componentDidMount() {
        new WOW({
            live: false
        }).init()
    }

    render() {
        const { summary } = this.props

        return (
            <PageSection color='#435465' height={'auto'}>
                <div className='row no-gutters'>
                    <div className='col-md-12 mt-2'>
                        <div className='row no-gutters justify-content-center m-3'>
                            <div className='col-md-8 text-center'>
                                <h2 className='text-white'>Creation and viewing</h2>
                            </div>
                        </div>

                        <div className='row no-gutters justify-content-center mb-2 text-white text-center'>
                            <div className='col-md-8 col-11'>
                                Use our components on any platform – in Win applications,
                                Web projects or JavaScript solutions. We support work with all major
                                technologies – WinForms, WPF, ASP.NET, MVC, .NET Core, JavaScript,
                                PHP, etc.
                            </div>
                        </div>

                        <div className='row no-gutters justify-content-center mb-2'>
                            <div className='col-md-10 col-lg-8 mt-4 mb-4 d-none d-md-block'>
                                <div className='row no-gutters'>
                                    {
                                        summary.map(v => (
                                            <div className='col-md-2' key={v.value}>
                                                <SummaryLabel
                                                    value={v.value}
                                                    label={v.label}
                                                    inverted={v.inverted}
                                                />
                                            </div>
                                        ))
                                    }
                                </div>
                            </div>
                        </div>

                        <div className='row no-gutters justify-content-center mb-5'>
                            <div className='col-10 col-md-7 col-lg-5'>
                                <LinkButton href='/dashboard'>Explore our features in the Online Demo</LinkButton>
                            </div>
                        </div>
                    </div>
                </div>
            </PageSection>
        )
    }
}

SummaryOverviewSection.propTypes = {
    summary: PropTypes.array.isRequired
}

export default SummaryOverviewSection