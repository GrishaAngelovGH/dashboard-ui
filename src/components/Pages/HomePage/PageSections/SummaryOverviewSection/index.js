import SummaryOverviewSection from './SummaryOverviewSection'

export default SummaryOverviewSection