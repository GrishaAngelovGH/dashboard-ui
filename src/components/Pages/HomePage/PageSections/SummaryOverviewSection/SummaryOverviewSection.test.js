import { render } from '@testing-library/react'
import { MemoryRouter } from 'react-router-dom'

import SummaryOverviewSection from './SummaryOverviewSection'

test('should render SummaryOverviewSection component', () => {
    const summary = [
        { value: '7+', label: 'supported platforms', inverted: false },
        { value: '17+', label: 'customizable elements', inverted: true }
    ]

    const view = render(<SummaryOverviewSection summary={summary} />, { wrapper: MemoryRouter })

    expect(view).toMatchSnapshot()
})