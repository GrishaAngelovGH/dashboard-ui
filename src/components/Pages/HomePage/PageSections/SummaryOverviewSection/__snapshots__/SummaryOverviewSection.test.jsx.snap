// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render SummaryOverviewSection component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row no-gutters"
        style="background: rgb(67, 84, 101); height: auto;"
      >
        <div
          class="col-md-12"
        >
          <div
            class="row no-gutters"
          >
            <div
              class="col-md-12 mt-2"
            >
              <div
                class="row no-gutters justify-content-center m-3"
              >
                <div
                  class="col-md-8 text-center"
                >
                  <h2
                    class="text-white"
                  >
                    Creation and viewing
                  </h2>
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-2 text-white text-center"
              >
                <div
                  class="col-md-8 col-11"
                >
                  Use our components on any platform – in Win applications, Web projects or JavaScript solutions. We support work with all major technologies – WinForms, WPF, ASP.NET, MVC, .NET Core, JavaScript, PHP, etc.
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-2"
              >
                <div
                  class="col-md-10 col-lg-8 mt-4 mb-4 d-none d-md-block"
                >
                  <div
                    class="row no-gutters"
                  >
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="d-flex flex-column wow fadeInUp m-2"
                        data-testid="summary-label"
                        data-wow-delay="0.5s"
                        data-wow-duration="5s"
                      >
                        <div
                          class="summary-value d-flex justify-content-center align-items-center"
                        >
                          7+
                        </div>
                        <div
                          class="summary-label text-center text-white text-uppercase m-1"
                        >
                          supported platforms
                        </div>
                      </div>
                    </div>
                    <div
                      class="col-md-2"
                    >
                      <div
                        class="d-flex flex-column flex-column-reverse wow fadeInUp m-2"
                        data-testid="summary-label"
                        data-wow-delay="0.5s"
                        data-wow-duration="5s"
                      >
                        <div
                          class="summary-value d-flex justify-content-center align-items-center"
                        >
                          17+
                        </div>
                        <div
                          class="summary-label text-center text-white text-uppercase m-1"
                        >
                          customizable elements
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="row no-gutters justify-content-center mb-5"
              >
                <div
                  class="col-10 col-md-7 col-lg-5"
                >
                  <a
                    class="d-block text-center text-uppercase link-button"
                    data-discover="true"
                    href="/dashboard"
                  >
                    Explore our features in the Online Demo
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row no-gutters"
      style="background: rgb(67, 84, 101); height: auto;"
    >
      <div
        class="col-md-12"
      >
        <div
          class="row no-gutters"
        >
          <div
            class="col-md-12 mt-2"
          >
            <div
              class="row no-gutters justify-content-center m-3"
            >
              <div
                class="col-md-8 text-center"
              >
                <h2
                  class="text-white"
                >
                  Creation and viewing
                </h2>
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-2 text-white text-center"
            >
              <div
                class="col-md-8 col-11"
              >
                Use our components on any platform – in Win applications, Web projects or JavaScript solutions. We support work with all major technologies – WinForms, WPF, ASP.NET, MVC, .NET Core, JavaScript, PHP, etc.
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-2"
            >
              <div
                class="col-md-10 col-lg-8 mt-4 mb-4 d-none d-md-block"
              >
                <div
                  class="row no-gutters"
                >
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="d-flex flex-column wow fadeInUp m-2"
                      data-testid="summary-label"
                      data-wow-delay="0.5s"
                      data-wow-duration="5s"
                    >
                      <div
                        class="summary-value d-flex justify-content-center align-items-center"
                      >
                        7+
                      </div>
                      <div
                        class="summary-label text-center text-white text-uppercase m-1"
                      >
                        supported platforms
                      </div>
                    </div>
                  </div>
                  <div
                    class="col-md-2"
                  >
                    <div
                      class="d-flex flex-column flex-column-reverse wow fadeInUp m-2"
                      data-testid="summary-label"
                      data-wow-delay="0.5s"
                      data-wow-duration="5s"
                    >
                      <div
                        class="summary-value d-flex justify-content-center align-items-center"
                      >
                        17+
                      </div>
                      <div
                        class="summary-label text-center text-white text-uppercase m-1"
                      >
                        customizable elements
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
            <div
              class="row no-gutters justify-content-center mb-5"
            >
              <div
                class="col-10 col-md-7 col-lg-5"
              >
                <a
                  class="d-block text-center text-uppercase link-button"
                  data-discover="true"
                  href="/dashboard"
                >
                  Explore our features in the Online Demo
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
